processor/spanmetricsprocessor/processor_test.go

```diff
@@ -289,10 +330,11 @@ func BenchmarkProcessorConsumeTraces(b *testing.B) {
 	}
 }
 
-func newProcessorImp(mexp *mocks.MetricsExporter, tcon *mocks.TracesConsumer, defaultNullValue *string) *processorImp {
+func newProcessorImp(mexp *mocks.MetricsExporter, tcon *mocks.TracesConsumer, defaultNullValue *string, temporality string) *processorImp {
```

Comment:
While we are here, can I ask a few things here?

- `ProcessorImp` is not go idiomatic and we should look to improve the naming of it, `processor` is a reasonable name for it.
- Pass in `testing.TB` so that the logger can be set to `zaptest.NewLogger(tb)`
- `defaultNullValue doesn't appear to be used, can it be removed?

Reply:
I'd prefer not to rename `processorImp` in this PR since it's the underlying structure that performs most operations in the processor. 

Why would we want to change from `zap.NewNop()` to `zaptest.NewLogger(tb)`? Is there a specific reason in this case? 

`defaultNullValue` is used, see line 356